Classes/Typo3/RefIndex.php

```diff
@@ -102,7 +101,7 @@ public function update()
     {
         // update index of selected tables
         foreach ($this->getSelectedTables() as $selectedTable) {
-            if (array_search($selectedTable, $this->getExistingTables()) !== false) {
+            if (in_array($selectedTable, $this->getExistingTables())) {
```

Comment:
I think I would add a third parameter to the in_array(x,y,true) to have it strict comparison on type.